frontend/src/components/AnswerButtonChoice.tsx

```diff
@@ -0,0 +1,106 @@
+import {Button} from "@mui/material";
+import React, {useEffect, useState} from "react";
+
+interface AnswerButtonChoiceProps {
+    animal_type: string
+    firstLetterOfAnimalName: string
+
+}
+
+export default function AnswerButtonChoice(props: AnswerButtonChoiceProps) {
+    const ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+    const [answer, setAnswer] = useState<boolean>(false);
+    const [firstRandomLetter, setFirstRandomLetter] = useState<string>("");
+    const [secondRandomLetter, setSecondRandomLetter] = useState<string>("");
+    const choices: string[] = [props.firstLetterOfAnimalName, firstRandomLetter, secondRandomLetter]
+    const [choicesShuffled, setChoicesShuffled] = useState<Array<string>>([]);
+
+    useEffect(() => {
+        setFirstRandomLetter(generateNewRandomLetter(ALPHABET,
+            props.firstLetterOfAnimalName));
+        }, [])
+
+    useEffect(() => {
+        setSecondRandomLetter(generateNewRandomLetter(ALPHABET,
+            props.firstLetterOfAnimalName,
+            firstRandomLetter));
+    }, [firstRandomLetter])
+
+
+    useEffect(() => {
+        setChoicesShuffled(shuffleArray(choices));
+    }, [secondRandomLetter])
+
+
+
+    /**
+     * returns a random Capital Letter of the Alphabet which is different from other provided letters
+     * @params ALPHABET CONST in Capital Letters, Letter from which the return value should differ from
+     * */
+    function generateNewRandomLetter(alphabet: string, letter: string, secondLetter?: string) {
+        const randomLetter = alphabet[Math.floor(Math.random() * alphabet.length)];
+        while (randomLetter === letter || randomLetter === secondLetter) {
+            const nextRandomLetter = alphabet[Math.floor(Math.random() * alphabet.length)];
+            if (nextRandomLetter !== letter && nextRandomLetter !== secondLetter) {
+                return nextRandomLetter;
+            }
+        }
+        return randomLetter;
+    }
```

Comment:
```suggestion
        return alphabet[Math.floor(Math.random() * alphabet.length)];
    }
```

Reply:
maybe this can become a bug, because now there is no check to find out whether this letter is already one of the options?

Reply:
If you pass only valid letters as options than not

Reply:
i think the comment explaining this is missing :/

Reply:
https://github.com/NiklasSchibel/AgameBC/pull/15#discussion_r800577510

But its only refactoring, what you did is fine :)